app/src/lib/find-default-branch.ts

```diff
@@ -0,0 +1,56 @@
+import { Branch, BranchType } from '../models/branch'
+import { Repository } from '../models/repository'
+import { getRemoteHEAD } from './git'
+import { getDefaultBranch } from './helpers/default-branch'
+
+/**
+ * Attempts to locate the default branch as determined by the HEAD symbolic link
+ * in the contribution target remote (origin or upstream) if such a ref exists,
+ * falling back to the value of the `init.defaultBranch` configuration and
+ * finally a const value of `main`.
+ *
+ * In determining the default branch we prioritize finding a local branch but if
+ * no local branch matches the default branch name nor is tracking the
+ * contribution target remote HEAD we'll fall back to looking for the remote
+ * branch itself.
+ */
+export async function findDefaultBranch(
+  repository: Repository,
+  branches: ReadonlyArray<Branch>
+) {
+  const originHead = await getRemoteHEAD(repository, 'origin')
+  const defaultBranchName = originHead ?? (await getDefaultBranch())
+  const remoteRef = originHead ? `origin/${originHead}` : undefined
```

Comment:
This should be `origin`/`upstream` depending on `getForkContributionTarget(repository)`, right?